perseus-api/perseus_api.py

```diff
@@ -25,83 +25,52 @@ def get_app_version():
-def load_saved_source_schema_call(current_user):
-    """load saved source schema by name"""
-    app.logger.info("REST request to load saved source schema")
-    try:
-        schema_name = request.args['schema_name']
-        saved_schema = load_saved_source_schema_from_server(current_user, schema_name)
-    except InvalidUsage as e:
-        raise e
-    except Exception as error:
-        raise InvalidUsage(error.__str__(), 500)
-    return jsonify([s.to_json() for s in saved_schema])
+        raise InvalidUsage('Schema was not loaded', 500)
+    return jsonify('OK')
 
 
-@perseus.route(f'/api/save_and_load_schema', methods=['GET', 'POST'])
+@perseus.route('/api/upload_scan_report_and_create_source_schema', methods=['POST'])
 @username_header
-def save_and_load_schema_call(current_user):
-    """save schema to server and load it from server in the same request"""
-    app.logger.info("REST request to save and load schema")
+def upload_scan_report_and_create_source_schema(current_user):
+    """Save White Rabbit scan report to server and create source schema"""
+    app.logger.info("REST request to upload WR scan report and create source schema")
     try:
         delete_generated_xml(current_user)
-        if request.method == 'POST':
-            file = request.files['file']
-            load_schema_to_server(file, current_user)
-        saved_schema = load_saved_source_schema_from_server(current_user, file.filename)
+        scan_report_file = request.files['scanReportFile']
+        load_scan_report_to_server(scan_report_file, current_user)
+        saved_schema = create_source_schema_by_scan_report(current_user, scan_report_file.filename)
     except InvalidUsage as error:
         raise error
     except Exception as error:
         raise InvalidUsage(error.__str__(), 500)
     return jsonify([s.to_json() for s in saved_schema])
 
 
-@perseus.route(f'/api/load_schema_to_server', methods=['POST'])
-@username_header
-def load_schema_call(current_user):
-    """save schema to server and load it from server in the same request"""
-    app.logger.info("REST request to load schema to server")
-    try:
-        file = request.files['file']
-        load_schema_to_server(file, current_user)
-    except Exception as error:
-        raise InvalidUsage('Schema was not loaded', 500)
-    return jsonify('OK')
-
-
-@perseus.route('/api/save_source_schema_to_db', methods=['POST'])
+@perseus.route('/api/create_source_schema', methods=['POST'])
 @username_header
-def save_source_schema_to_db_call(current_user):
-    app.logger.info("REST request to save source schema to db")
+def create_source_schema(current_user):
+    """Create source schema by source tables from ETL mapping"""
+    app.logger.info("REST request to create source schema")
     try:
         source_tables = request.json
-        save_source_schema_in_db(current_user, source_tables)
+        create_source_schema_in_db(current_user, source_tables)
     except Exception as error:
```

Comment:
What kind of exceptions can be raised? Would it be better to handle each one separately?